src/Impostor.Server/Net/Inner/Objects/Components/InnerPlayerPhysics.cs

```diff
@@ -1,22 +1,35 @@
 ﻿using System;
 using System.Threading.Tasks;
 using Impostor.Api;
+using Impostor.Api.Events.Managers;
 using Impostor.Api.Net;
 using Impostor.Api.Net.Messages;
 using Impostor.Server.Net.State;
+using Impostor.Server.Events.Player;
 using Microsoft.Extensions.Logging;
 
 namespace Impostor.Server.Net.Inner.Objects.Components
 {
     internal partial class InnerPlayerPhysics : InnerNetObject
     {
         private readonly ILogger<InnerPlayerPhysics> _logger;
+        private readonly IEventManager _eventManager;
         private readonly InnerPlayerControl _playerControl;
+        private readonly Game _game;
+        private bool _isWatchingCamera;
+        public bool IsWatchingCamera { get {return _isWatchingCamera; }
+            internal set {
+                _isWatchingCamera = value;
+                _eventManager.CallAsync(new PlayerWatchingCameraEvent(_game, _game.GetClientPlayer(_playerControl.OwnerId), _playerControl, value));
```

Comment:
Please DO NOT call an async method in a setter of a property.

Reply:
Are there any suggestions on how to control `IsWatchingCamera` and call Event?

Reply:
Create an async setter method in the classes to change the state.